mlir/lib/Dialect/SCF/Utils/Utils.cpp

```diff
@@ -500,6 +503,8 @@ static bool areInnerBoundsInvariant(scf::ForOp forOp) {
 }
 
 /// Unrolls and jams this loop by the specified factor.
+/// This function doesn't verify that the loop is parallel, if there are true
+/// loop carried dependencies, this function will produce invalid code.
 LogicalResult mlir::loopUnrollJamByFactor(scf::ForOp forOp,
```

Comment:
Good comment above!
apart from moving the resposbility to the user, should we have minimal checks about the usage of the `iter_args`? are there cases we can always block?

Reply:
In the typical use case, I believe the user will be in a better position to decide whether the loop is parallel or not. For instance, because the loop comes from a `parallel` iterator in a `linalg` op, or because it's the result of a tiling operation. We could try to detect some individual cases with ad-hoc checks, but we'd be introducing overhead for arguably little to no utility beyond avoiding an easy to notice typo (or "temporary lapse in judgement"), and the transformation would still not be safe.

If somebody else wants to start adding analysis to filter-out invalid cases, I won't stop them, but I think at least we should match functionality with the source counterpart in `affine`.

Reply:
I see that in affine there is handling for `iter_args`  - please look at https://github.com/llvm/llvm-project/blob/12762b8648d88e6823aedc82f663a596fa2eef76/mlir/lib/Dialect/Affine/Utils/LoopUtils.cpp#L1134

The handling is for reductions there but still it is a case we can address / block